client/src/components/record/video-player.tsx

```diff
@@ -106,46 +96,72 @@ function VideoPlayer(props: {
     >
       <VideoRecorder
         classes={classes}
-        curAnswer={curAnswer}
-        recordState={recordState}
         height={height}
         width={width}
-        onRecordStart={onRecordStart}
-        onRecordStop={onRecordStop}
+        recordState={recordState}
       />
       <div
         style={{
           position: "absolute",
-          visibility: curAnswer.videoSrc ? "visible" : "hidden",
+          visibility:
+            recordState.curAnswer!.videoSrc ||
+            recordState.curAnswer?.isUploading
+              ? "visible"
+              : "hidden",
         }}
       >
         <Typography
           data-cy="warning"
           style={{
             textAlign: "center",
             visibility:
-              curAnswer.videoSrc &&
-              videoLength < (curAnswer.minVideoLength || 0)
+              recordState.curAnswer!.videoSrc &&
```

Comment:
let's not use the `not-null` assertion anywhere, most lint checkers consider this an error

Reply:
fd9f53b